pkg/component/ai/embedding.go

```diff
@@ -0,0 +1,39 @@
+package ai
+
+type EmbeddingInput struct {
```

Comment:
nit: please, document exported types. Although it doesn't produce a `golangci-lint` error / warning, many IDEs do show a warning here (and it hurts my eyes 🥲).